packages/@lwc/integration-karma/test/component/face-callbacks/index.spec.js

```diff
@@ -0,0 +1,122 @@
+    const form = container.shadowRoot.querySelector('form');
+    return form;
+};
+
+const sanityTestFormAssociatedCustomElements = (ctor) => {
+    let form;
+    let face;
+
+    beforeEach(() => {
+        form = createFormElement();
+        face = createElement('face-form-associated', { is: ctor });
+        form.appendChild(face);
+    });
+
+    it('cannot access formAssociated outside of a component', () => {
+        expect(() => face.formAssociated).toLogErrorDev(
+            'Error: [LWC error]: formAssociated cannot be accessed outside of a component. Set the value within the component class.'
+        );
+    });
+
+    it('calls face lifecycle methods', () => {
+        // formAssociatedCallback
+        expect(face.formAssociatedCallbackHasBeenCalled).toBeTruthy();
+
+        // formDisabledCallback
+        face.setAttribute('disabled', true);
+        expect(face.formDisabledCallbackHasBeenCalled).toBeTruthy();
+
+        // formResetCallback
+        form.reset();
+        expect(face.formResetCallbackHasBeenCalled).toBeTruthy();
+
+        // Note there is no good way to test formStateRestoreCallback in karma tests
+    });
+
+    it('is associated with the correct form', () => {
+        const form2 = document.createElement('form');
+        form2.setAttribute('class', 'form2');
+        const face2 = createElement('face-form-associated-2', { is: ctor });
+        form2.appendChild(face2);
+        const container = document.body.querySelector('face-container');
+        container.shadowRoot.appendChild(form2);
+
+        expect(face.internals.form.className).toEqual('form1');
+        expect(face2.internals.form.className).toEqual('form2');
+    });
+};
+
+const testErrorThrownWhenStaticFormAssociatedNotSet = (ctor) => {
+    it(`throws an error if 'static formAssociated' is not set`, () => {
+        const form = createFormElement();
+        const notFormAssociated = createElement('face-not-form-associated', {
+            is: ctor,
+        });
+
+        expect(() => form.appendChild(notFormAssociated)).toThrowConnectedError(
+            `Form associated lifecycle methods must have the 'static formAssociated' value set in the component's prototype chain.`
+        );
+    });
+};
+
+if (window.lwcRuntimeFlags.ENABLE_NATIVE_CUSTOM_ELEMENT_LIFECYCLE) {
+    describe('native lifecycle', () => {
+        if (process.env.NATIVE_SHADOW) {
+            describe('native shadow', () => {
+                sanityTestFormAssociatedCustomElements(FormAssociated);
+                testErrorThrownWhenStaticFormAssociatedNotSet(NotFormAssociated);
+            });
+        } else {
+            describe('synthetic shadow', () => {
+                it('cannot be used and throws an error', () => {
+                    const form = createFormElement();
+                    const face = createElement('face-form-associated', { is: FormAssociated });
+
+                    expect(() => form.appendChild(face)).toThrowConnectedError(
```

Comment:
Looks like the face callbacks throw errors in the same way as native `connectedCallback`/`disconnectedCallback` so I reused the `toThrowConnectedError`, maybe we should rename it to something like `toThrowNativeCallbackError`.